src/SonarScanner.MSBuild.PreProcessor/ProcessedArgs.cs

```diff
@@ -135,6 +135,8 @@ public string PropertiesFileName
             HttpTimeout = TimeoutProvider.HttpTimeout(AggregateProperties, logger);
         }
 
+        protected /* for testing */ ProcessedArgs() { }
```

Comment:
Sorry for suggesting to remove it. Now that you know how it's used could you please:
- remove the comment
- add a new one explaining why exactly it is needed.

Reply:
There are a dozen `/* for testing */` comments in various forms in the code base. I think the pattern is common enough in S4NET to learn it. Also, we have a couple of rules, that respect the [DynamicallyAccessedMembersAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.dynamicallyaccessedmembersattribute?view=net-8.0) or a custom attribute of the same name. We should add this attribute to signal "reflection-only access". I don't think such an effort belongs here, as we would need to 
* Create the custom attribute
* Annotate all places where there is a re-flection only usage